### PR TITLE
Make cardano-node work with new diffusion arguments

### PR DESCRIPTION
**Pull request by bolt12**
Here is the PR that introduces the changes to `DiffusionArguments` in `ouroboros-network`: https://github.com/input-output-hk/ouroboros-network/pull/3079

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Run.hs [hunk 1]
```diff
@@ -439,12 +432,16 @@ createDiffusionArguments NodeConfiguration {
...
   case nt of
     RealNodeTopology lrpg prp _ ->
-      concatMap (rootAddressToRelayAddress . localRoots) (groups lrpg)
-      ++ concatMap rootAddressToRelayAddress (map publicRoots prp)
+      ( map (\lrp -> (valency lrp
+                           , Map.fromList $ rootAddressToRelayAddress
+                                          $ localRoots lrp))
+                  (groups lrpg)
+      , concatMap (map fst . rootAddressToRelayAddress)
+                  (map publicRoots prp))
```
Comment (coot) [inline cardano-node/src/Cardano/Node/Run.hs:444]:
```suggestion
       ( map (\lrp -> ( valency lrp
                      , Map.fromList $ rootAddressToRelayAddress
                                     $ localRoots lrp
                      )
             )
             (groups lrpg)
      , concatMap (map fst . rootAddressToRelayAddress)
                  (map publicRoots prp)
      )
```

**Review by coot — CHANGES_REQUESTED** [review]

**Comment by bolt12** [comment]
@coot is this one good to go?

File: cardano-node/src/Cardano/Node/Run.hs [hunk 2]
```diff
@@ -439,12 +432,19 @@ createDiffusionArguments NodeConfiguration {
...
-      ++ concatMap rootAddressToRelayAddress (map publicRoots prp)
+       ( map (\lrp -> ( valency lrp
+                      , Map.fromList $ rootAddressToRelayAddress
+                                     $ localRoots lrp
+                      )
+             )
+             (groups lrpg)
+      , concatMap (map fst . rootAddressToRelayAddress)
+                  (map publicRoots prp)
+      )
```
Comment (coot) [inline cardano-node/src/Cardano/Node/Run.hs:447]: I think this is still misaligned by one space :).

Comment (bolt12) [reply cardano-node/src/Cardano/Node/Run.hs:447]: I'll do it by hand then 😛

**Review by coot — APPROVED** [review]
LGTM

**Pull request merged**